src/runtime_src/core/include/xgq_impl.h

```diff
@@ -41,6 +41,7 @@
 #if defined(__KERNEL__)
 # include <linux/types.h>
 #else
+# include <stdbool.h>
```

Comment:
warning: including `stdbool.h` has no effect in C++; consider removing it [hicpp-deprecated-headers,modernize-deprecated-headers]

```cpp
# include <stdbool.h>
~~~~~~~~~~^~~~~~~~~~~
```

Reply:
If this header is supposed to be included from C++ code in addition to C code, then maybe fix the include section for C++ within #ifdef __cplusplus ?

Reply:
> 
> 
> If this header is supposed to be included from C++ code in addition to C code, then maybe fix the include section for C++ within #ifdef __cplusplus ?

Will try it.